sushichef.py

```diff
@@ -40,18 +41,27 @@ def construct_channel(self, **kwargs):
             book_info['absolute_url'] = page_url + book_info['relative_url']
 
             # Add book to channel tree
-            topic_node_source_id = 'topic/' + book_info['subject']
-            page_topic_node = TopicNode(title=book_info['subject'], source_id=topic_node_source_id)
-            channel.add_child(page_topic_node)
-            doc_node = DocumentNode(
-                title=book_info['title'],
-                description=book_info['description'],
-                source_id=book_info['source_id'],
-                license=get_license('CC BY-SA', copyright_holder='Stack Overflow'),
-                language='en',
-                files=[DocumentFile(path=book_info['absolute_url'], language='en')],
-            )
-            page_topic_node.add_child(doc_node)
+            book_node_source_id = 'topic/' + book_info['subject']
+            book_node = TopicNode(title=book_info['subject'], source_id=book_node_source_id)
+            channel.add_child(book_node)
+
+            # Get chapters info
+            pdf_path = book_info['absolute_url']
+            with PDFParser(pdf_path) as pdfparser:
+                chapters = pdfparser.split_chapters()
+
+            # Add chapter nodes
+            for chapter in chapters:
+                chapter_node_source_id = book_info['source_id'] + '/' + chapter['title']
+                chapter_node = DocumentNode(
+                    title=chapter['title'],
+                    description=chapter['title'],
```

Comment:
We have that info in the `title` already so no need to repear.

A better alternative would be to set the description to something like `Chapter n of the Goalkicker book on book_info['subject']` which contains the parent book title, this way is a teacher uses just one chapter of the book in another context, readers will still know where to get the other chapeters.